cf_xarray/accessor.py

```diff
@@ -2209,7 +2209,7 @@ def add_bounds(self, keys: str | Iterable[str], *, dim=None):
                 apply_mapper(_get_all, self._obj, key, error=False, default=[key])
             )
 
-        obj = self._maybe_to_dataset(self._obj.copy(deep=True))
+        obj = self._maybe_to_dataset(self._obj.copy(deep=False))
```

Comment:
Let's make sure the bounds aren't associated with the original dataset.

Reply:
I think this is now covered in `test_add_bounds`.